x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichCache.java

```diff
@@ -26,38 +22,37 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
-import java.util.Objects;
 import java.util.concurrent.atomic.AtomicLong;
-import java.util.function.BiConsumer;
+import java.util.function.Consumer;
 import java.util.function.LongSupplier;
 import java.util.function.ToLongBiFunction;
 
 /**
  * A simple cache for enrich that uses {@link Cache}. There is one instance of this cache and
  * multiple enrich processors with different policies will use this cache.
  * <p>
- * The key of the cache is based on the search request and the enrich index that will be used.
- * Search requests that enrich generates target the alias for an enrich policy, this class
- * resolves the alias to the actual enrich index and uses that for the cache key. This way
- * no stale entries will be returned if a policy execution happens and a new enrich index is created.
- * <p>
  * There is no cleanup mechanism of stale entries in case a new enrich index is created
  * as part of a policy execution. This shouldn't be needed as cache entries for prior enrich
  * indices will be eventually evicted, because these entries will not end up being used. The
  * latest enrich index name will be used as cache key after an enrich policy execution.
- * (Also a cleanup mechanism also wouldn't be straightforward to implement,
+ * (Also a cleanup mechanism wouldn't be straightforward to implement,
  * since there is no easy check to see that an enrich index used as cache key no longer is the
- * current enrich index the enrich alias of an policy refers to. It would require checking
+ * current enrich index that the enrich alias of a policy refers to. It would require checking
  * all cached entries on each cluster state update)
  */
 public final class EnrichCache {
 
+    private static final CacheValue EMPTY_CACHE_VALUE = new CacheValue(List.of(), CacheKey.CACHE_KEY_SIZE);
+    /**
+     * This field defines the size that the enrich cache takes up on the heap without any entries in it.
+     */
+    private static final int BASE_ENRICH_CACHE_SIZE = Cache.NUMBER_OF_SEGMENTS * 152;
```

Comment:
When I added this, I realized that this might be confusing to users when there are no entries in the enrich cache but the enrich stats say it's taking up heap space. I'm thinking we could either
1. subtract this value when we return the stats -- to keep the internal representation accurate but avoid confusing the user 
2. remove this base value as it's only `38KB` which is pretty small
3. leave it like this and accept that some users might be slightly confused.

What do you think?

Reply:
I'd rather revert this commit and avoid the complexity for now -- we can continue this conversation another time, but I don't want to overcomplicate this PR.